packages/connector-tencent-sms/src/http.ts

```diff
@@ -0,0 +1,109 @@
+import type { BinaryToTextEncoding } from 'crypto';
+import crypto from 'crypto';
+
+import got from 'got';
+
+import type { TencentErrorResponse, TencentSuccessResponse } from '@/schema';
+import { tencentErrorResponse } from '@/schema';
+
+const endpoint = 'sms.tencentcloudapi.com';
+
+function sha256Hmac(message: string, secret: string): string;
+function sha256Hmac(message: string, secret: string, encoding: BinaryToTextEncoding): Buffer;
+
+function sha256Hmac(message: string, secret: string, encoding?: BinaryToTextEncoding) {
+  const hmac = crypto.createHmac('sha256', secret);
+
+  return encoding ? hmac.update(message).digest(encoding) : hmac.update(message).digest();
```

Comment:
Why using `? and :`, seems no difference combining the two.
```suggestion
  return hmac.update(message).digest(encoding);
```

Reply:
The digest parameter of the crypto typescript definition is not allowed to be undefined

Reply:
But isn't this equivalent to `undefined`?

<img width="237" alt="image" src="https://user-images.githubusercontent.com/12833674/208609891-29dad4c8-864e-4b18-a573-3f90a13fe519.png">

Reply:
I see. Then this is OK